doc/overview/data_integrity.md

```diff
@@ -1,67 +1,38 @@
-
-There are two key requirements that DAOS will support.
-
- 1. Detect silent data corruption - Corruption will be detected on the
- distribution and attribute keys and records within a DAOS object. At a minimum,
- when corruption is detected, an error will be reported.
-1. Correct data corruption - When data corruption is detected, an attempt will
- be made to recover the data using data redundancy mechanisms.
-
-### Supportive/Additional Requirements
-
-Additionally, DAOS will support:
-
- 1. End to End Data Integrity as a Quality of Service Attribute - Container
- properties are used to enable/disable the use of checksums for data integrity
- as well as define specific attributes of data integrity feature.  Refer to
-[Data Integrity Readme](https://daos-stack.github.io/user/container/#data-integrity)
-for details on configuring a container with checksums enabled.
-
- 1. Minimize Performance Impact - When there is no data corruption, the End to
- End Data Integrity feature should have minimal performance impacted. If data
- corruption is detected, performance can be impacted to correct the data.
- Work is ongoing to minimize performance impact.
-1. Inject Errors - The ability to corrupt data within a specific record, key,
- or checksum will be necessary for testing purposes. Fault injection is used to
- simulate corruption over the network and on disk. The `DAOS_CSUM_CORRUPT_*`
- flags used for data corruption are defined in `src/include/daos/common.h`.
-1. Logging - When data corruption is detected, error logs are captured in
- the client and server logs.
-
-Features not yet supported:
-
- 1. Event Logging - When silent data corruption is discovered, an event should
- be logged in such a way that it can be retrieved with other system health and
- diagnostic information.
-1. Proactive background service task - A background task on
- the server which scans for and detects (audits checksums) silent data
- corruption and corrects.
+DAOS uses checksums internally to discover silent data corruption. While each
+component in a system (network layer, storage devices) may offer protection
+against silent data corruption, DAOS provides end-to-end data integrity to
+better protect user data. If silent data corruption is detected, DAOS will
+attempt to recover the corrupted data using data redundancy mechanisms
+(Replication or Erasure Code).
+
+## End-to-end Data Integrity
+
+In simple terms, end-to-end means that the DAOS Client library will calculate a
+checksum for data that is being sent to the DAOS Server. The DAOS Server will
+store the checksum and return it upon data retrieval. Then the client verifies
+the data by calculating a new checksum and comparing to the checksum received
+from the server. There are variations on this approach depending on the type of
+data being protected, but the following diagram shows the basic checksum flow.
+![Basic Checksum Flow](../graph/data_integrity/basic_checksum_flow.png)
+
+- [ ] Performance Impact
```

Comment:
performance impact, break up into oview, dev....
What is the intent here? Does not look like it fits with graphic.

Reply:
I meant to remove that. This PR has already landed so I'll create a follow up change to fix that.